Feature/setup testing

Sam09040 opened the pull request:
setup of the tests

Sam09040 committed:
Working user add to bd

## add:
- password hash
- prisma client functions
- many users and user by id queries

Sam09040 committed:
Working user add to bd

## add:
- password hash
- prisma client functions
- many users and user by id queries

Sam09040 committed:
Add password validation

## add:
- password validation function and add it to the resolver

Sam09040 committed:
Add password validation

## add:
- password validation function and add it to the resolver

Sam09040 committed:
test setup

- test\index.mjs file and basic code
- test script

Sam09040 committed:
test setup

## add:
- test\index.mjs file and basic code
- test script

Sam09040 committed:
hello query test

## add:
- hello on query
- axios library
- hello query test

Sam09040 committed:
hello query test

- hello on query
- axios library
- hello query test

racsandoval reviewed (commented):
Esse PR está mostrando arquivos da branch anterior

Sam09040 committed:
Small changes

## update:
- index.mjs to index.test.mjs
- include all files in test and src directories

Sam09040 committed:
Small changes

## update:
- index.mjs to index.test.mjs
- include all files in test and src directories

Sam09040 committed:
Small changes

Sam09040 committed:
Small changes

Sam09040 committed:
File organization

## add:
- graphql folder under src
- schema.graphql under graphql
- resolvers.ts under graphql

## update:
- small change on index.ts

Sam09040 committed:
File organization

## add:
- graphql folder under src
- schema.graphql under graphql
- resolvers.ts under graphql

## update:
- small change on index.ts

Sam09040 committed:
Merge branch 'feature/db-user-add' of https://github.com/Sam09040/botActivities into feature/db-user-add

Sam09040 committed:
Merge pull request #12 from Sam09040/feature/db-user-add

Add user to database

Sam09040 committed:
Merge branch 'feature/setup-testing' of https://github.com/Sam09040/botActivities into feature/setup-testing

racsandoval commented on .prettierrc.js:
pq tem dois arquivos pro prettier?

Sam09040 replied:
honestamente eu não sei. em algum momento um segundo arquivo apareceu e eu não me atentei a isso, então vou apagá-lo.

racsandoval commented on src/graphql/resolvers.ts:
```diff
@@ -0,0 +1,77 @@
+
+export const resolvers = {
+  Query: {
+    hello: () => "hello!",
+    test: () => "it's working!",
```
essas duas queries não tem o mesmo propósito?

Sam09040 replied:
tem sim, vou apagar uma delas pois é redundante ter ambas.

racsandoval commented on test/index.test.mjs:
pq está usando a extensão `mjs` ao invés de `ts`

Sam09040 replied:
inicialmente eu queria usar a extensão ts, mas estava dando muitos erros, como que não conseguia encontrar os módulos, não conseguia entender a extensão ts, entre outros. eu tentei muitas formas de tentar resolver isso, seja usando o ts-nodes, arquivo mocharc, entre outros, e inclusive jonathas tentou me ajudar também, mas nada deu efeito, portanto, eu optei por usar a extensão mjs pois me assegura o uso do ESModules e o mocha consegue chegar nele e lê-lo sem problemas.

racsandoval replied:
entendi, acho uma boa insistir um pouco mais em fazer os arquivos TS funcionarem, sem eles perdemos a tipagem e algumas outras funcionalidades do typescript

Sam09040 committed:
Corrections

## remove:
- .prettierrc.ts file
- test string from schema and resolver

Sam09040 committed:
requested changes

- update resolvers import
- change test file from .mjs to .ts
- add tsimp to run ts files

walteraandrade commented on src/graphql/resolvers.ts:
```diff
@@ -0,0 +1,76 @@
+import bcrypt from 'bcrypt';
+const prisma = new PrismaClient();
+
+const isPasswordValid = (password: string): boolean => {
```
essa função parece fora do lugar dela nesse arquivo 🤔

Sam09040 replied:
você tem um ótimo ponto! obrigada por apontar isso!

walteraandrade commented on src/index.ts:
```diff
@@ -1,40 +1,15 @@
-});
+export default server;
+
+/* server.listen().then(async ({ url }) => {
```
por que esse comentário ficou aqui? 🤔

Sam09040 replied:
eu fiz esse comentário, pois eu importava a variável `server` do index.ts, logo essa parte rodava também e impactava nos testes. já criei um arquivo de exportação a parte para isso e um não impacta mais no outro.

Sam09040 committed:
corrections

## add:
- server export file;
- password validation separated from resolver;

## update:
- corrections on test file;
